lms/djangoapps/grades/models.py

```diff
@@ -239,11 +244,17 @@ def save_grade(cls, **kwargs):
         user_id = kwargs.pop('user_id')
         usage_key = kwargs.pop('usage_key')
         try:
-            grade, is_created = cls.objects.get_or_create(user_id=user_id, usage_key=usage_key, defaults=kwargs)
+            grade, is_created = cls.objects.get_or_create(
+                user_id=user_id,
+                course_id=usage_key.course_key,
+                usage_key=usage_key,
+                defaults=kwargs,
+            )
         except IntegrityError:
-            is_created = False
-        if not is_created:
-            grade.update(**kwargs)
+            cls.update_grade(user_id=user_id, usage_key=usage_key, **kwargs)
+        else:
+            if not is_created:
+                grade.update(**kwargs)
```

Comment:
Double-check my logic in this block:

get_or_create:
- if get is successful, no exception raised, fall through to `grade.update`
- if create is successful, no exception raised, no further action taken
- if create fails due to integrity issue, `cls.update_grade` b/c it will exist